File: minimumpath.py
class Solution:
    def minFallingPathSum(self, A: List[List[int]]) -> int:
        #initializing DP matrix
        dp=[[0 for i in range(len(A))]for j in range(len(A))]
        #checking condition for null value
        if A==None or len(A)==0:
            return 0
        #initializing first row of DP matrix
        dp[0]=A[0]
        for i in range(1,len(A)):
            for j in range(len(A)):
                #updating the first column of DP matrix
                if j==0:
                    dp[i][j]=A[i][0]+min(dp[i-1][0],dp[i-1][1])
                #updating the last column of DP matrix
                elif j==len(A)-1:
                    dp[i][j]=A[i][len(A)-1]+min(dp[i-1][len(A)-1],dp[i-1][len(A)-2])
                #updating the remaining colums of DP matrix
                else:
                    dp[i][j]=A[i][j]+min(dp[i-1][j-1],dp[i-1][j+1],dp[i-1][j])
        return min(dp[len(A)-1])

File: minimumpath.java
/*

// Time Complexity : O(n *m) where n is number of row and m is number of column.
// Space Complexity : O(n *m) where n s number of row and m is number of column.
// Did this code successfully run on Leetcode : Yes

*/

public class minimumpath {

    public static int minFallingPathSum(int[][] matrix) {
        int n = matrix.length;
        int m = matrix[0].length;
        int[][] dp = new int[n][m];
        for (int i = 0; i < m; i++) {

            dp[0][i] = matrix[0][i];
        }
        for (int i = 1; i < n; i++) {

            dp[i][0] += matrix[i][0] + Math.min(dp[i - 1][0], dp[i - 1][1]);

            dp[i][m - 1] += matrix[i][m - 1] + Math.min(dp[i - 1][m - 1], dp[i - 1][m - 2]);

            for (int j = 1; j < m - 1; j++) {

                dp[i][j] += matrix[i][j] + Math.min(dp[i - 1][j], Math.min(dp[i - 1][j - 1], dp[i - 1][j + 1]));
            }
        }

        int ans = Integer.MAX_VALUE;

        for (int i = 0; i < m; i++) {

            ans = Math.min(ans, dp[n - 1][i]);
        }
        return ans;
    }
}